overlay/make-package-set/full.nix

```diff
@@ -11,7 +11,7 @@
 {
   packageFun,
   workspaceSrc ? null,
-  rustChannel,
+  rustToolchain,
```

Comment:
Maybe I am not understanding the change correctly, but shouldn't this work?

```nix
{
  description = "Workspace flake";

  inputs = {
    cargo2nix.url = "github:cargo2nix/cargo2nix?ref=be-friendly-to-users";
    flake-compat = {
      url = "github:edolstra/flake-compat";
      flake = false;
    };
    flake-utils.url = "github:numtide/flake-utils";
    nixpkgs.url = "github:nixos/nixpkgs?ref=release-21.05";
    rust-overlay.url = "github:oxalica/rust-overlay";
    rust-overlay.inputs.nixpkgs.follows = "nixpkgs";
    rust-overlay.inputs.flake-utils.follows = "flake-utils";
  };

  outputs = { self, nixpkgs, cargo2nix, flake-utils, rust-overlay, ... }:
    flake-utils.lib.eachDefaultSystem (system:
      let
        overlays = [ (import "${cargo2nix}/overlay") rust-overlay.overlay ];
        pkgs = import nixpkgs {
          inherit overlays system;
        };

        rustPkgs = pkgs.rustBuilder.makePackageSet' {
          packageFun = import ./Cargo.nix;
          rustToolchain= "nightly-2021-12-05";
        };

        workspaceShell = rustPkgs.workspaceShell {};

      in
      rec {
        devShell = workspaceShell;
      }
    );
}
```

Reply:
`full.nix` is only called by `simplified.nix`.   Sorry for the confusing arrangement.  The `rustToolchain` here is an internal result.  When you call `makePackageSet`` it calls into `simplified.nix` first.

You want to look at the changes to simplified.nix to understand the new logic.  The argument documentation was also updated in the root flake.nix.

passing `rustToolchain` completely overrides the toolchain.  Cargo2nix will just trust that whatever you gave it in `rustToolchain` contains rustc, cargo, std, and rust-src components at the usual paths.  This is full user control with no hand holding.

The normal user will now just pass `rustVersion` and get a `stable.${rustVersion}.minimal...` toolchain from rust-overlay

For nightly, you want to pass `rustChannel = "nightly"` and `rustVersion=YYYY-MM-DD`.  Search for `selectLatestNightlyWith` to see exactly how this gets handled after the changes.

Finally, I've added a combined overlay, so you can completely omit the `rust-overlay` inputs in your example and just instantiate nixpkgs with `cargo2nix.overlay`

Reply:
Thanks for the clarifications!